### ISSUE TITLE
Random failures of "[CONNECTED] == true" condition. How to debug?

### ISSUE THREAD

**Issue by Linkandzelda** [TwiN/gatus#361]
### Describe the bug

I've setup Gatus to use ICMP to ping my internal VMs by their dns name from one of the VMs. Everything works fine, except randomly they all report a failure 1 time on the condition `[CONNECTED] == true` all at the same time. After that, they are all green again for a couple of hours until they fail again once. One of those VMs is the actual VM Gatus is running on, and that one shows failed as well.

### What do you see?

I see the status as `[CONNECTED] (false) == true` as failed

### What do you expect to see?

Given that I'm unable to find an issue with the internal network I expect to see `[CONNECTED] == true`.

### List the steps that must be taken to reproduce this issue

1. Setup an ICMP endpoint with `[CONNECTED] == true` on a bridged VM network interface

### Version

stable

### Additional information

I have a few TCP endpoints that also fail at the same time as the ICMP ones.

**Comment by Linkandzelda** [comment]
Wanted to add, I have a few TCP endpoints that also fail at the same time as the ICMP ones.

**Comment by TwiN** [comment]
I've run Gatus on hundreds of machines, different environments and several cloud providers, and I can tell you with some degree of confidence that this type of problem is usually due to networking or hardware issues.

You could try to run a script that pings indefinitely and see if when Gatus has the issue you mentioned, the ping script also has this issue.

**Comment by Linkandzelda** [comment]
Thank you for the comment. What I've ran is this snippet for testing:
`while true; do ping vmnodename-1 -c 1; sleep 0.1; done`

This was to test DNS resolution and ping at a fast rate. What I discovered is that my DNS randomly reports NXDOMAIN. So I believe that was able to point me in the direction that this is not a Gatus issue.

Thank you for your fast response, and thank you for creating this software!
